aspnet/mvc/overview.rst

```diff
@@ -1,8 +1,313 @@
+
+ASP.NET Core MVC in ASP.NET Core includes support for building web pages and HTTP services in a single aligned framework that can be hosted in IIS or self-hosted in your own process.
+
+Features
+--------
+
+ASP.NET Core MVC includes the following features:
+  - `Routing`_
+  - `Model binding`_
+  - `Model validation`_
+  - `Dependency injection`_
+  - `Filters`_
+  - `Areas`_
+  - `Web APIs`_
+  - `Testability`_
+  - `Razor view engine`_
+  - `Strongly typed views`_
+  - `Mobile support`_
+
+Routing
+^^^^^^^
+
+ASP.NET Core MVC is built on top of :doc:`ASP.NET Core's routing </fundamentals/routing>`, a powerful URL-mapping component that lets you build applications that have comprehensible and searchable URLs. This enables you to define your application's URL naming patterns that work well for search engine optimization (SEO) and for link generation, without regard for how the files on your web server are organized. You can define your routes using a convenient route template syntax that supports route value constraints, defaults and optional values.
+
+*Convention-based routing* enables you to globally define the URL formats that your application accepts and how each of those formats maps to a specific action method on given controller. When an incoming request is received, the routing engine parses the incoming URL and matches it to one of the defined URL formats, and calls the associated controller's action method.
+
+
+.. code-block:: c#
+
+  routes.MapRoute(name: "Default", template: "{controller=Home}/{action=Index}/{id?}");
+
+*Attribute routing* enables you to specify routing information by decorating your controllers and actions with attributes that define your application's routes. This means that your route definitions are placed next to the controller and action with which they're associated.
+
+.. code-block:: c#
+  :emphasize-lines: 1,4
+
+  [Route("api/[controller]")]
+  public class ProductsController : Controller
+  {
+    [HttpGet("{id}")]
+    public IActionResult GetProduct(int id)
+    {
+      ...
+    }
+  }
+
+Model binding
+^^^^^^^^^^^^^
+
+ASP.NET Core MVC :doc:`model binding </mvc/models/model-binding>` converts client request data  (form values, route data, query string parameters, HTTP headers) into objects that the controller can handle. As a result, your controller logic doesn't have to do the work of figuring out the incoming request data; it simply has the data as parameters to its action methods.
+
+.. code-block:: C#
+
+  public async Task<IActionResult> Login(LoginViewModel model, string returnUrl = null) { ... }
+
+Model validation
+^^^^^^^^^^^^^^^^
+
+In addition to binding request data to model objects, ASP.NET Core MVC provides a powerful :doc:`validation <mvc/models/validation>` system. Simply add validation attributes to your model types using data annotation attributes and then check for model errors in your controller action.
```

Comment:
ASP.NET Core MVC supports  :doc:`validation <mvc/models/validation>` by decorating your model object with  data annotation validation attributes. The validation attributes are check on the client side before values are posted to the server.